lib/langchain/tool/file_system/file_system.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Langchain::Tool
+  class FileSystem < Base
+    #
+    # A tool that wraps the Ruby file system classes.
+    #
+    # Usage:
+    #    file_system = Langchain::Tool::FileSystem.new
+    #
+    NAME = "file_system"
+    ANNOTATIONS_PATH = Langchain.root.join("./langchain/tool/#{NAME}/#{NAME}.json").to_path
+
+    description <<~DESC
+      A tool that wraps the Ruby File class, for interacting with the file system.
+    DESC
+
+    def list_directory(directory_path:)
+      Dir.entries(directory_path)
+    rescue Errno::ENOENT
+      "No such directory: #{directory_path}"
+    end
+
+    def read_file(file_path:)
+      File.read(file_path)
+    rescue Errno::ENOENT
+      "No such file: #{file_path}"
+    end
+
+    def write_to_file(file_path:, content:)
+      File.write(file_path, content)
```

Comment:
Do we want to handle error scenarios here, like permission denied? And should that apply to the other methods as well?

Reply:
Yeah, I just haven't been able to generate all the different error scenarios. Idk if it's all `Errno::ENOENT` that's raised.